soc/xtensa/intel_adsp/tools/hw-service.py

```diff
@@ -0,0 +1,449 @@
+
+        self.lock.acquire()
+        self.available = False
+        self.occpied_user = user
+        self.passcode = passcode
+        self.lock.release()
+        return True
+
+    def set_dev_unlock(self, user, passcode):
+        if self.lock_enable:
+            self.lock.acquire()
+            self.available = True
+            self.occpied_user = None
+            self.passcode = None
+            self.lock.release()
+
+    def is_dev_available(self, user):
+        if not self.lock_enable:
+            return True
+
+        if self.occpied_user == user or self.available:
+            return True
+        else:
+            return False
+
+    def set_fw_ready(self, fw_recv):
+        if fw_recv:
+            self.fw_queue.put(fw_recv)
+
+    def is_fw_ready(self):
+        self.fw_file = self.fw_queue.get()
+        log.info(f"Current FW is {self.fw_file}")
+
+        if self.fw_file:
+            if self.fw_file == "OTHER":
+                self.fw_file = None
+            return True
+        else:
+            return False
+
+    def cleanup(self):
+        self.lock.acquire()
+        self.script = None
+        if self.fw_file:
+            os.remove(self.fw_file)
+        self.fw_file = None
+        self.lock.release()
+
+    def get_script(self):
+        if os.geteuid() != 0:
+            self.script = ([f'sudo', f'{self.load_cmd}'])
+        else:
+            self.script = ([f'{self.load_cmd}'])
+
+        if self.is_load_fw:
+            self.script.append(f'{self.fw_file}')
+
+        if self.board.params:
+            for param in self.board.params:
+                self.script.append(param)
+
+        log.info(f'run script: {self.script}')
+        return self.script
+
+class board_config():
+    def __init__(self, board, serial, load_cmd):
+
+        board_intel_adsp = ['intel_adsp_cavs15', 'intel_adsp_cavs18',
+                            'intel_adsp_cavs20', 'intel_adsp_cavs20_jsl',
+                            'intel_adsp_cavs25', 'intel_adsp_cavs25_tgph']
+
+        board_x86_hypervisor = ['acrn_ehl_crb', 'acrn']
+
+        self.board = board          # the board name
+        self.serial = serial        # serial port using
+        self.load_cmd = load_cmd    # cmd for loading
+        self.params = []            # params of loading cmd
+        self.load_fw = False        # need to upload FW
+
+        if not self.board or self.board in board_intel_adsp:
+            log.info(f"Set board to intel_adsp by default.")
+            self.load_fw = True
+            if not self.load_cmd:
+                self.load_cmd = "./cavstool.py"
+
+        elif self.board in board_x86_hypervisor:
+            if not load_cmd:
+                self.load_cmd = "./acrn-test-pty.exp"
+
+        else:
+            if not self.serial:
+                log.error(f"Cannot find serial {self.serial}")
+                sys.exit(1)
+
+            if not self.load_cmd:
+                self.load_cmd = "/usr/bin/picocom"
```

Comment:
This dependency should be listed somewhere. I think most of the project uses `minicom`, can't you use it to be more aligned with the rest ?

Reply:
This is for boards which output log to the serial port. You are right, and I will using the python's serial instead of calling a picocom or minicom. (And it has remove from this PR, the support for other boards will be added in another PR)